src/librustc_mir/transform/generator.rs

```diff
@@ -1110,6 +1128,10 @@ fn create_cases<'tcx>(
                             Rvalue::Use(Operand::Move(resume_arg.into())),
                         )),
                     });
+                    statements.push(Statement {
+                        source_info,
+                        kind: StatementKind::StorageDead(resume_arg),
+                    });
```

Comment:
I don't think you should add a `StorageDead` for the argument here since arguments don't usually use `Storage*` instructions and there's no matching `StorageLive`.